src/badge/withBadge.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+
+import Badge from './Badge';
+
+const withBadge = (value, options = {}) => WrappedComponent => {
+  class WithBadge extends React.Component {
+    render() {
+      const {
+        left = -7,
+        bottom = 0,
+        top = -2,
+        right = 0,
+        hidden = false,
+        ...badgeProps
+      } = options;
+
+      const badgeValue =
+        typeof value === 'function' ? value(this.props) : value;
+
+      return (
+        <View style={styles.container}>
+          <WrappedComponent {...this.props} />
+
+          {!hidden && (
+            <Badge
+              badgeStyle={styles.badge}
```

Comment:
Lastly lets remove these default styles (badgeStyle, textStyle). That way we can use the default values from the Badge itself.

Also it would allow for the mini badge to be rendered.
![screen shot 2018-12-09 at 4 09 23 pm](https://user-images.githubusercontent.com/5962998/49702288-0117bb00-fbcd-11e8-98ec-4d73898bd274.jpg)

```jsx
const BadgedAvatar = withBadge(null, {
  right: -30,
  status: 'success',
})(Avatar);

<BadgedAvatar size="medium" title="MD" rounded />
```

Reply:
Sounds good!